include/secp256k1_silentpayments.h

```diff
@@ -0,0 +1,405 @@
+ *  payment.
+ *
+ *  If necessary, the private keys are negated to enforce the right y-parity.
+ *  For that reason, the private keys have to be passed in via two different
+ *  parameter pairs, depending on whether the seckeys correspond to x-only
+ *  outputs or not.
+ *
+ *  Returns: 1 if creation of outputs was successful. 0 if an error occured.
+ *  Args:                ctx: pointer to a context object
+ *  Out:   generated_outputs: pointer to an array of pointers to xonly pubkeys,
+ *                            one per recipient.
+ *                            The order of outputs here matches the original
+ *                            ordering of the recipients array.
+ *  In:           recipients: pointer to an array of pointers to silent payment
+ *                            recipients, where each recipient is a scan public
+ *                            key, a spend public key, and an index indicating
+ *                            its position in the original ordering. The
+ *                            recipient array will be sorted in place, but
+ *                            generated outputs are saved in the
+ *                            `generated_outputs` array to match the ordering
+ *                            from the index field. This ensures the caller is
+ *                            able to match the generated outputs to the
+ *                            correct silent payment addresses. The same
+ *                            recipient can be passed multiple times to create
+ *                            multiple outputs for the same recipient.
+ *              n_recipients: the number of recipients. This is equal to the
+ *                            total number of outputs to be generated as each
+ *                            recipient may passed multiple times to generate
+ *                            multiple outputs for the same recipient
+ *       outpoint_smallest36: serialized smallest outpoint (lexicographically)
+ *                            from the transaction inputs
+ *           taproot_seckeys: pointer to an array of pointers to 32-byte
+ *                            private keys of taproot inputs (can be NULL if no
+ *                            private keys of taproot inputs are used)
+ *         n_taproot_seckeys: the number of sender's taproot input private keys
+ *             plain_seckeys: pointer to an array of pointers to 32-byte
+ *                            private keys of non-taproot inputs (can be NULL
+ *                            if no private keys of non-taproot inputs are
+ *                            used)
+ *           n_plain_seckeys: the number of sender's non-taproot input private
+ *                            keys
+ */
+SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_silentpayments_sender_create_outputs(
+    const secp256k1_context *ctx,
+    secp256k1_xonly_pubkey **generated_outputs,
+    const secp256k1_silentpayments_recipient **recipients,
```

Comment:
Is there any reason not to make the outer pointer constant since it is modified in this function?

```suggestion
    const secp256k1_silentpayments_recipient * const *recipients,
```

Reply:
The recipients array is passed to `secp256k1_hsort`, where the array is sorted in place.